reactioncommerce/reaction
# (refactor): 3551 replace vsivsi:job-collection for npm module @reactioncommerce/job-queue

Resolves #3551
Impact: **breaking**
Type: **refactor**
 
## Issue
 
Deprecate the usage of the atmosphere package [visivisi:meteor-job-collection](https://github.com/vsivsi/meteor-job-collection) and move to an `npm` package https://github.com/reactioncommerce/reaction-job-queue.

This is to further reduce Reaction's dependency on atmosphere packages, and `vsivsi:job-collection` has been deprecated since `01-01-2018`.

## Solution
 
1. Convert from CoffeeScript to es2015 using http://decaffeinate-project.org/
1. Run through the files and fix linting issues, formatting
1. Fix warnings from the conversion provided by decaffeinate
1. Import where needed
1. Include any required `npm` packages like `later.js`
1. Convert all tests. 1 test file exists in the `@reactioncommerce/job-queue` package while the Meteor specific tests exist in the main app.
 
## Breaking changes
 
- `visivisi:meteor-job-collection` removed. Noe uses npm package https://github.com/reactioncommerce/reaction-job-queue
- New import path for both Job abd Job Collection. `import { Job, JobCollection } from "/imports/plugins/core/job-collection/server/jobCollection.js"`
- `mrt:later` removed due to removal of `visivisi:meteor-job-collection`
- Uses an npm package https://github.com/reactioncommerce/reaction-job-queue
- `Job` and `JobCollection` are now required to be instantiated with `new`. `new Job()` `new JobCollection()`
 
## Testing
 
**Test 1**
1. Run tests using `reaction test`, make sure they pass
1. Verify that email is properly configured
1. Add items to cart and checkout
1. Receive email

**Test 2**
1. Break email config so email is not configured
1. Add items to cart and checkout
1. Should not receive email
1. Go back to email settings and fix email config
1. Resend email by using the refresh button, next to the failed email
1. Receive email
